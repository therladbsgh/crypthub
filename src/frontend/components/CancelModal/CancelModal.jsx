import * as _ from 'lodash';
import React, { Component } from 'react';
import { Modal, Button, Message } from 'semantic-ui-react';
import { GameBackend } from 'endpoints';
import { CancelModalStyle as styles } from 'styles';

export default class CancelModal extends Component {
    constructor(props) {
        super(props);

		this.state = {
            open: false,
			loading: false,
			err: ''
		};
    
        this.open = this.open.bind(this);
        this.close = this.close.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    open() {
        this.setState({ open: true });
    }

    close() {
        this.setState({
            open: false,
			loading: false,
			err: ''
        });
    }

	handleSubmit(event) {
        this.setState({
            loading: true,
            err: ''
        });

		GameBackend.cancelOrder(this.props.tradeId)
		.then(res => {
			console.log('success! ', res);
            this.close();
		}, ({ err }) => {
			console.log('error! ', err);
			this.setState({ loading: false, err });
        });
    }
    
    render() {
        const { open, loading, err } = this.state;

        return (
            <Modal trigger={<Button compact negative size='small' content='Cancel' />} open={open} onOpen={this.open} onClose={this.close} closeIcon>
                <Modal.Header id={styles.cancel}>Confirm Cancellation</Modal.Header>
                    <Modal.Content>
                        <p>Are you sure you want to cancel this trade order?</p>
                        {err && <Message error header='Error' content={err} />}
                    </Modal.Content>
                <Modal.Actions>
                    <Button negative icon='check' content='Cancel Order' onClick={this.handleSubmit} loading={loading} />
                </Modal.Actions>
            </Modal>
        );
    }
}
