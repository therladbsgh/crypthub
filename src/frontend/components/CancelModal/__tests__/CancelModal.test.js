import React from 'react';
import CancelModal from '../CancelModal';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  	const tree = renderer.create(<CancelModal />).toJSON();
  	expect(tree).toMatchSnapshot();
});