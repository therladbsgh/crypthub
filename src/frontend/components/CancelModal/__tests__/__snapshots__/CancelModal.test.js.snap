// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="ui tiny compact negative button"
  disabled={undefined}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="button"
  tabIndex={undefined}
>
  <i
    aria-hidden="true"
    className="cancel icon"
  />
  Cancel
</button>
`;
